src/main/java/org/spongepowered/api/item/inventory/BlockCarrier.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.util.Direction;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.Optional;
+
+/**
+ * Represents a Block that is a carrier of an Inventory
+ * e.g. An Anvil
+ */
+public interface BlockCarrier extends Carrier {
+
+    /**
+     * Returns the Location of the Block.
+     *
+     * @return The Location of the Block
+     */
+    Location<World> getLocation();
+
+    /**
+     * Returns the Inventory at given location if owned by this Carrier
+     *
+     * @param at The location
+     * @return The inventory at given location
+     */
+    Optional<Inventory> getInventory(Location<World> at);
+
+    /**
+     * Returns the Inventory at given location if owned by this Carrier and
+     * accessible from given direction.
+     *
+     * e.g. A Furnace accessed from {@link Direction#DOWN} will return its Fuel Slot
+     *
+     * @param at The Location
+     * @param from The Direction
+     * @return The inventory at the location when accessed from given direction
+     */
+    Optional<Inventory> getInventory(Location<World> at, Direction from);
```

Comment:
I would suggest these methods:
`Optional<Inventory> getInventory(Direction from);`
This can be in block carrier, with the location implied to be the called carrier's location.
(I suggest leaving my method in this class and moving two methods above to `MultiBlockCarrier`)